File: js/20_fila.js
const Queue = require('./lib/queue.js');

const fila = new Queue(); // Cria uma fila vazia

// Insere algumas pessoas na fila
fila.enqueue('Leotildes');
fila.enqueue('Orozimbo');
fila.enqueue('Valdisney');
fila.enqueue('Adamastor');

const atendido = fila.dequeue();
console.log(`Atendido: ${atendido}`);
console.log(fila.toString());

fila.enqueue('Marcinéia');
console.log(fila.toString());

const proximo = fila.peek();
console.log(`Próximo a ser atendido: ${proximo}`);
console.log(fila.toString());

File: python/20_fila.py
from lib.queue import Queue

fila = Queue()  # Cria uma fila vazia

# Insere algumas pessoa na fila
fila.enqueue('Leotildes')
fila.enqueue('Orozimbo')
fila.enqueue('Valdisney')
fila.enqueue('Adamastor')

atendido = fila.dequeue()
print(f"Atendido: {atendido}")
print(fila)

fila.enqueue('Marcinéia')
print(fila)

proximo = fila.peek()
print(f"Próximo a ser atendido: {proximo}")
print(fila)
